elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexWorkVisitor.java

```diff
@@ -132,8 +133,15 @@ public ElasticsearchIndexWorkVisitor(String indexName, boolean refreshAfterWrite
 
 	@Override
 	public BackendRequest<?> visitFlushWork(FlushLuceneWork work, IndexingMonitor monitor) {
-		// Nothing to do
-		return null;
+		Flush flush = new Flush.Builder().addIndex( indexName ).refresh( true ).build();
```

Comment:
From what I can read in [the documentation](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/indices-flush.html), since we're not setting `wait_if_ongoing` to `true` here, the operation will fail if a flush operation is in progress. I'd say it's not what we want, especially given that we're already executing the operation asynchronously and thus waiting a bit more is no big deal.